src/main/java/application/service/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package application.service;
+
+import application.dto.shopping.cart.ShoppingCartResponseDto;
+import application.mapper.CartItemMapper;
+import application.mapper.ShoppingCartMapper;
+import application.model.CartItem;
+import application.model.ShoppingCart;
+import application.model.User;
+import application.repository.ShoppingCartRepository;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public ShoppingCart save(ShoppingCart shoppingCart) {
+        return shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart findByUserId(Long id) {
+        return shoppingCartRepository.findShoppingCartByUserId(id);
+    }
+
+    @Override
+    public Set<CartItem> updateSetCartItem(Long id, ShoppingCart s, CartItem c) {

```

Comment:
Name variables and params properly

Reply:
Also, method name `updateSet...` is confusing, seems like this method does few things.